dev/how-to/troubleshoot/tidb-in-kubernetes.md

```diff
@@ -18,3 +56,206 @@ TiDB Operator 使用 PV (Persistent Volume)/PVC (Persistent Volume Claim) 来存
+    kubectl -n ${namespace} -it exec ${pod_name} -- cat /etc/pd/pd.toml | grep client-urls
+    # TiKV
+    kubectl -n ${namespace} -it exec ${pod_name} -- cat /etc/tikv/tikv.toml | grep addr
+    # TiDB
+    kubectl -n ${namespace} -it exec ${pod_name} -- cat /etc/tidb/tidb.toml | grep port
+    ```
+
+## Pod 处于 Pending 状态
+
+Pod 处于 Pending 状态，通常都是资源不满足导致的，比如：
+
+* 使用持久化存储的 PD、TiKV、Monitor Pod 使用的 PVC 的 StorageClass 不存在或 PV 不足
+* Kubernetes 集群中没有节点能满足 Pod 申请的 CPU 或内存
+
+此时，可以通过 `kubectl describe pod` 命令查看 Pending 的具体原因：
+
+{{< copyable "shell-regular" >}}
+
+```
+kubectl describe po -n ${namespace} ${pod_name}
+```
+
+如果是 CPU 或内存资源不足，可以通过降低对应组件的 CPU或内存资源申请使其能够得到调度，或是增加新的 Kubernetes 节点。
+
+如果是 PVC 的 StorageClass 找不到，则需要将 TiDB 删除，并且将对应 PVC 也都删除，然后在 `values.yaml` 里面将 `StorageClassName` 修改为集群中可用的 StorageClass 名字，可以通过下面命令获取集群中可用的 StorageClass：
+
+{{< copyable "shell-regular" >}}
+
+```
+kubectl get storageclass
+```
+
+如果集群中有 StorageClass，但可用的 PV 不足，则需要添加对应的 PV 资源。对于 Local PV，可以参考 [本地 PV 配置](/reference/configuration/tidb-in-kubernetes/local-pv-configuration.md) 进行扩充。
+
+## Pod 处于 CrashLoopBackOff 状态
+
+Pod 处于 CrashLoopBackOff 状态意味着 Pod 内的容器重复地异常退出（异常退出后，容器被 Kubelet 重启，重启后又异常退出，如此往复）。可能导致 CrashLoopBackOff 的原因有很多，此时，最有效的定位办法是查看 Pod 内容器的日志：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl -n ${namespace} logs -f ${pod_name}
+```
+
+假如本次日志没有能够帮助诊断的有效信息，可以添加 -p 参数输出容器上次启动时的日志信息:
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl -n ${namespace} logs -p ${pod_name}
+```
+
+确认日志中的错误信息后，可以根据 [tidb-server 启动报错](/how-to/troubleshoot/cluster-setup.md#tidb-server-启动报错)，[tikv-server 启动报错](/how-to/troubleshoot/cluster-setup.md#tikv-server-启动报错)，[pd-server 启动报错](/how-to/troubleshoot/cluster-setup.md#pd-server-启动报错) 中的指引信息进行进一步排查解决。
+
+另外，TiKV 在 ulimit 不足时也会发生启动失败的状况，对于这种情况，可以修改 Kubernetes 节点的 `/etc/security/limits.conf` 调大 ulimit：
+
+```
+root        soft        nofile        1000000
+root        hard        nofile        1000000
+root        soft        core          unlimited
+root        soft        stack         10240
+```
+
+假如通过日志无法确认失败原因，ulimit 也设置正常，那么可以通过 [诊断模式](#诊断模式) 进行进一步排查。
+
+## 无法访问 TiDB 服务
+
+TiDB 服务访问不了时，首先确认 TiDB 服务是否部署成功，确认方法如下：
+
+查看该集群的所有组件是否全部都启动了，状态是否为 Running。
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl get po -n ${namespace}
+```
+
+检查 TiDB 组件的日志，看日志是否有报错。
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl logs -f ${tidb-pod-name} -n ${namespace}
+```
+
+如果确定集群部署成功，则进行网络检查：
+
+1. 如果你是通过 `NodePort` 方式访问不了 TiDB 服务，请在 node 上尝试使用 service domain 或 clusterIP 访问 TiDB 服务，假如 serviceName 或 clusterIP 的方式能访问，基本判断 Kubernetes 集群内的网络是正常的，问题可能出在下面两个方面：
+
+    * 客户端到 node 节点的网络不通。
+    * 查看 TiDB service 的 externalTrafficPolicy 属性是否为 Local。如果是 Local 则客户端必须通过 TiDB Pod 所在 node 的 IP 来访问。
+
+2. 如果 service domain 或 clusterIP 方式也访问不了 TiDB 服务，尝试用 TiDB服务后端的 `${PodIP}:4000` 连接看是否可以访问，如果通过 PodIP 可以访问 TiDB 服务，可以确认问题出在 service domain 或 clusterIP 到 PodIP 之间的连接上，排查项如下：
+
+    * 检查 dns 服务是否正常：
+
+    {{< copyable "shell-regular" >}}
```

Comment:
Since this code block is under `* 检查 DNS 服务是否正常：`, please indent it with 4 more spaces. Otherwise, the display on the PingCAP website is not what we want. It's the same case for the following several code blocks.

Reply:
get it